Updates to Backup & Recovery section (#892)
BitcoinDesign/Guide: guide/daily-spending-wallet/backup-and-recovery/recovery.md

# Wallet recovery

> There are several reasons why a user might need to restore their bitcoin wallet. They may have lost their device or accidentally deleted their wallet application. Perhaps they have simply bought a new phone and wish to migrate their wallet over. Regardless of why, it’s important that your wallet application has an easy way for users to recover their funds.

Reviewer: ```suggestion
There are several reasons why a user might need to restore their bitcoin wallet. They may have lost their phone or accidentally deleted their wallet application. Perhaps they have simply bought a new phone and wish to migrate their wallet over. Regardless of why, it’s important that your wallet application has an easy way for users to recover their funds.
```

Use phone instead of device so its more in-line with the scope of this reference design

> This wallet is designed to make the recovery process as easy as possible for the user. It can pull the user's channel state from cloud backup, allowing the user to pick up right where they left off. It will also sync contacts and metadata from the cloud backup, so there is no loss of useful transaction history. The private key can be pulled from a cloud backup or manually provided by the user.

Reviewer: ```suggestion
This wallet is designed to make the recovery process as easy as possible for the user. It can pull the user's channel state from the cloud backup, allowing the user to pick up right where they left off. It will also sync contacts and metadata from the cloud backup, so there is no loss of useful transaction history. The recovery phrase can be pulled from a cloud backup or manually provided by the user.
```

Reviewer: Lets avoid saying private key and use recovery phrase instead for simplicity sake, and what our readers (designers) would be more familiar with. Technically recovery phrase is actually more accurate in this scenario too.

   layout = "shadow"
   caption = "The wallet has been successfully restored."
   alt-text = "Screen showing a home screen with funds"
   width = 250
   height = 541
%}

</div>

## Recovering with a manual backup

This involves a few more steps, but should work just as well as the auto cloud backup. This works great for [basic recovery scenarios](#basic-recovery). In the manual restore, the primary difference is that the app asks the user to type in their recovery phrase. It will attempt to automatically pull their lightning channel state from cloud backup.

<div class="image-slide-gallery">

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-manual.png"
   retina = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-manual@2x.png"
   modalImage="/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-manual@2x.png"
   layout = "shadow"
   caption = "User selects \"advanced\" from the restore screen, and is taken here."
   alt-text = "Screen showing description of manual restore"
   width = 250
   height = 541
%}

{% include picture.html

>    image = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-manual-recovery-phrase-progress.png"

Reviewer 2: We should add auto-complete here, as it's enough to enter the first few characters usually.

Author: Let's discuss this here before I make another commit. My initial thought is that it should be separate and distinct from the OS level autocomplete. The whole advantage is that it should pull from the limited 2048 word BIP39 dictionary.

We can show one screen where we use autocorrection to prevent an error. The below is similar to the existing screen, but we are instead detecting if the word is wrong before the user even finishes typing it. There's no `poi` words after `point` in the dict, so the app can assume `poil` is a typo.

![restore-manual-recovery-phrase-error](https://user-images.githubusercontent.com/2414177/184961861-4aa38074-aac9-47b2-8f30-b8351e4926da.png)

Then we can show an example of where autocomplete prevents the user from having to type in a long word. In this case, `hospital`, because there are no other `hosp` words in the dict.

![restore-manual-recovery-phrase-progress](https://user-images.githubusercontent.com/2414177/184962231-01df735e-fe4f-483b-9661-e7a879e4fd33.png)

With the above picture, I was thinking about how the user takes advantage of the autocomplete. Do they just tap the word to have it fill automatically? Would they know to do this? What if the autocorrect stood out more? So I made the below image, which would be a little popup that shows the closest guess. The popup is the same color as the primary button, so I think it would stand out as something you interact with.

![restore-manual-recovery-phrase-progress-1](https://user-images.githubusercontent.com/2414177/184962465-09f4365e-67cd-4df5-b71c-a62df059043a.png)

Another option might be to have a custom autocomplete bar above the keyboard, but that might look kind of tacky in place of the OS autocomplete.

Thoughts?

Reviewer: I like the OS autocomplete, as long as it only uses words from the word list that is. Also its only required to enter the first 4 letters for each entry, could mention this.

Reviewer 2: Stephen and I discussed yesterday. Option 2 seems the way to go, as OS autocompletes don't allow for custom dictionaries. URL bars also have a similar approach to option 2.

Author: @Bosch-0 I decided against the OS autocomplete. It's doesn't matter to me if feeding the OS autocomplete a custom dictionary is possible or not, because the user may not understand this. Seeing words from their recovery phrase start appearing in the autocomplete might feel somewhat alarming, like the phone is tracking their bitcoin activity or something.

Instead, I added something that I hope feels more familiar to how autocomplete would work in a mobile web browser.

Reviewer 2: I like the new design you went for, except for the light blue, doesn't seem to go with everything else.

   width = 250
   height = 541
%}

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-restoring.png"
   retina = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-restoring@2x.png"
   modalImage="/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-restoring@2x.png"
   layout = "shadow"
   caption = "App begins to restore, pulling the channels backup automatically from the cloud."
   alt-text = "Screen showing an animation as the wallet is being restored"
   width = 250
   height = 541
%}

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-restored.png"
   retina = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-restored@2x.png"
   modalImage="/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-restored@2x.png"
   layout = "shadow"
   caption = "Wallet is finished being restored."
   alt-text = "Screen showing that the wallet has been restored."
   width = 250
   height = 541
%}

{% include picture.html

>    image = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-finish.png"

Reviewer 2: This screen uses the old home screen design that shows the balance. Maybe show the activity screen that shows balance and transactions - as a sign that it was all restored?

Author: Done

%}

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-manual-cloud-ui.png"
   retina = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-manual-cloud-ui@2x.png"
   modalImage="/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-manual-cloud-ui@2x.png"
   layout = "shadow"
   caption = "User authenticats with the cloud provider, finds the backup file, etc."
   alt-text = "Screen showing a placeholder image for a cloud provider UI"
   width = 250
   height = 541
%}

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-manual-checking-file.png"
   retina = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-manual-checking-file@2x.png"
   modalImage="/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-manual-checking-file@2x.png"
   layout = "shadow"
   caption = "The app checks the backup file the user has selected."
   alt-text = "Screen showing a backup file being checked"
   width = 250
   height = 541
%}

{% include picture.html

>    image = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-manual-checking-complete.png"

Reviewer 2: This image looks identical to the previous one...?

Author: The other says "checking", this one says "looks good". I can strip out the other one if this is too much detail.

Reviewer 2: Ah, I see it now. Maybe the second one could have green text and a checkmark or something like that?

Author: Done

   modalImage="/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-manual-pending-notice@2x.png"
   layout = "shadow"
   caption = "App reminds the user of the delay in accessing funds."
   alt-text = "Screen describing the delay in accessing funds"
   width = 250
   height = 541
%}

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-manual-finish-pending.png"
   retina = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-manual-finish-pending@2x.png"
   modalImage="/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-manual-finish-pending@2x.png"
   layout = "shadow"
   caption = "The delay is also indicated on the homescreen of the app."
   alt-text = "Screen showing the wallet homescreen with pending   indicators"
   width = 250
   height = 541
%}

</div>

### Handling emergency scenarios

In a situation where the app provider or LSP is offline or uncooperative, the user will have no assistance in getting their funds. While this situation is unlikely, it should be prepared for.

> This wallet offers an open source wallet recovery tool. This tool is available on GItHub and linked to from their website. In the event that the app is taken down from app stores or the LSP's node is down for an extended period of time, the user can use their backup to force close their channels and have all their bitcoin sent to an on-chain address.

Reviewer 2: ```suggestion
This wallet offers an open source wallet recovery tool. This tool is available on GitHub and linked to from their website. In the event that the app is taken down from app stores or the LSP's node is down for an extended period of time, the user can use their backup to force close their channels and have all their bitcoin sent to an on-chain address.
```

> This wallet offers an open source wallet recovery tool. This tool is available on GitHub and linked to from their website. In the event that the app is taken down from app stores or the LSP's node is down for an extended period of time, the user can use their backup to force close their channels and have all their bitcoin sent to an on-chain address.

Reviewer: You could have this in an Emergency section in the settings that simply uses a SCB to force close all channels, a separate tool is only necessary if the user can't even access the app at all which is the most extreme emergency scenario. 

Phoenix has this in their app for an example to look at.

Author: There's already a section about force closing within the app itself. Its 3 lines above the section you are commenting on.

![Screen Shot 2022-08-19 at 7 32 38 AM](https://user-images.githubusercontent.com/2414177/185609660-e8c8bb31-8008-40be-ad93-e44da1c1f3f6.png)

## Recovering with an automatic cloud backup

This is by far the simplest way to recover the wallet. The user opts to restore from their cloud backup. The app checks the cloud account for a backup and restores the lightning node on the user's device. This works great for [basic recovery scenarios](#basic-recovery).

<div class="image-slide-gallery">

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-menu.png"
   retina = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-menu@2x.png"
   modalImage="/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-menu@2x.png"
   layout = "shadow"

>    caption = "Beginning of the restore flow."

Reviewer 2: This button is taller than the orange buttons elsewhere.

Author: Fixed

   width = 250
   height = 541
%}

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-restored.png"
   retina = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-restored@2x.png"
   modalImage="/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-restored@2x.png"
   layout = "shadow"
   caption = "Wallet is finished being restored."
   alt-text = "Screen showing that the wallet has been restored."
   width = 250
   height = 541
%}

{% include picture.html
   image = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-finish.png"
   retina = "/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-finish@2x.png"
   modalImage="/assets/images/guide/daily-spending-wallet/backup-and-recovery/recovery/restore-finish@2x.png"
   layout = "shadow"
   caption = "The wallet has been successfully restored."
   alt-text = "Screen showing a home screen with funds"
   width = 250
   height = 541
%}

</div>

{% include tip/open.html color="green" icon="check" label="Use autocomplete" %}

> This wallet makes the recovery process easier for the user by autocompleting the words of their recovery phrase as they type. Since recovery phrases are comprised of words from a predefined dictionary of 2,048 words, the app is able usually able to autocomplete or detect a typo with only 3 or 4 letters typed.

Reviewer 2: ```suggestion
This wallet makes the recovery process easier for the user by autocompleting the words of their recovery phrase as they type. Since recovery phrases are comprised of words from a predefined dictionary of 2,048 words, the app is usually able to autocomplete or detect a typo with only 3 or 4 letters typed.
```

{% include tip/close.html %}

## Other situations

### Cloud backup screws up

> While this wallet will attempt to pull channel state from a cloud backup, there may be situations in which this does not work automatically. A simple example would be if the user setup the wallet on iOS but is migrating to Android. This could be remedied by allowing the user to specify another cloud account to use.

Reviewer 2: ```suggestion
While this wallet will attempt to pull channel state from a cloud backup, there may be situations in which this does not work automatically. A simple example would be if the user set up the wallet on iOS, but is migrating to Android. This could be remedied by allowing the user to specify another cloud account to use.
```
